weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/ObjectViewerSection.tsx

```diff
@@ -153,6 +157,18 @@ const ObjectViewerSectionNonEmpty = ({
     setExpandedIds(getGroupIds());
   };
 
+  // On first render and when data changes, recompute expansion state
+  useEffect(() => {
+    const isSimple = isSimpleData(data);
+    const newMode = isSimple || isExpanded ? 'expanded' : 'collapsed';
+    if (newMode === 'expanded') {
+      onClickExpanded();
+    } else {
+      onClickCollapsed();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data, isExpanded]);
```

Comment:
q: what are we ignoring here? could we make those functions callbacks?

Reply:
It's ignoring onClickExpanded and onClickCollapsed - not sure about making them callbacks, they both use mode state and set it, I think you would just end up with two different exhaustive-deps disablings?

Reply:
right... 👍🏻